decidim-proposals/app/controllers/decidim/proposals/admin/proposals_controller.rb

```diff
@@ -124,13 +124,26 @@ def update_scope
         def edit
           enforce_permission_to :edit, :proposal, proposal: proposal
           @form = form(Admin::ProposalForm).from_model(proposal)
-          @form.attachment = form(AttachmentForm).from_params({})
+          document = proposal.documents.first
+          @form.attachment = if document.present?
+                               form(AttachmentForm).from_params({ file: document.file, title: translated_attribute(document.title) })
+                             else
+                               form(AttachmentForm).from_params({})
+                             end
         end
 
         def update
           enforce_permission_to :edit, :proposal, proposal: proposal
 
           @form = form(Admin::ProposalForm).from_params(params)
+
+          if params[:proposal][:attachment].present?
+            check_delete_file = params[:proposal][:attachment][:delete_file]
+            attachment_id = params[:proposal][:attachment][:id].to_i
+          end
+
+          Attachment.find(params[:proposal][:attachment][:id].to_i).delete if check_delete_file == "1" && attachment_id == proposal.documents.first.id
```

Comment:
This logic should belong to the `Admin::UpdateProposal` command. Also check the `Decidim::AttachmentMethod` and commands handling similar features, since they might be useful!